golos.publication/golos.publication.cpp

```diff
@@ -466,16 +466,13 @@ void publication::close_message(structures::mssgid message_id) {
 
     asset mssg_reward;
     auto state = pool->state;
-    eosio_assert(mssg_itr->max_payout.symbol == state.funds.symbol, "LOGIC ERROR! publication::payrewards: mssg_itr->max_payout.symbol != state.funds.symbol");
     int64_t payout = 0;
     if(state.msgs == 1) {
         payout = state.funds.amount; //if we have the only message in the pool, the author receives a reward anyway
-        payout = std::min(payout, mssg_itr->max_payout.amount);
         eosio_assert(state.rshares == mssg_itr->state.netshares, 
                 "LOGIC ERROR! publication::payrewards: pool->rshares != mssg_itr->netshares for last message");
         eosio_assert(state.rsharesfn == sharesfn.data(), 
                 "LOGIC ERROR! publication::payrewards: pool->rsharesfn != sharesfn.data() for last message");
-        state.funds.amount = 0;
```

Comment:
In case of WebAssemble it is better to minimize the number of calls to the chain.
So, it is better to return back the previous code without subtractions and `eosio_assert()`-s